osf/migrations/0171_migrate_draft_registration_perms.py

```diff
@@ -0,0 +1,78 @@
+    INSERT INTO auth_group (name)
+    SELECT regexp_split_to_table('draft_registration_' || D.id || '_read,draft_registration_' || D.id || '_write,draft_registration_' || D.id || '_admin', ',')
+    FROM osf_draftregistration D;
+    """
+
+remove_draft_auth_groups = """
+    DELETE FROM auth_group WHERE name in
+    (SELECT regexp_split_to_table('draft_registration_' || D.id || '_read,draft_registration_' || D.id || '_write,draft_registration_' || D.id || '_admin', ',')
+    FROM osf_draftregistration D);
+    """
+
+# Forward migration - add read permissions to all draft registration django read groups, add read/write perms
+# to all draft registration django write groups, and add read/write/admin perms to all draft registration django admin groups
+add_permissions_to_draft_registration_groups = """
+    -- Adds "read_draft_registration" permissions to all Draft Reg read groups - uses DraftRegistrationGroupObjectPermission table
+    INSERT INTO osf_draftregistrationgroupobjectpermission (content_object_id, group_id, permission_id)
+    SELECT D.id as content_object_id, G.id as group_id, PERM.id AS permission_id
+    FROM osf_draftregistration AS D, auth_group G, auth_permission AS PERM
+    WHERE G.name = 'draft_registration_' || D.id || '_read'
+    AND PERM.codename = 'read_draft_registration';
+
+    -- Adds "read_draft_registration" and "write_draft_registration" permissions to all Draft Reg write groups
+    INSERT INTO osf_draftregistrationgroupobjectpermission (content_object_id, group_id, permission_id)
+    SELECT D.id as content_object_id, G.id as group_id, PERM.id AS permission_id
+    FROM osf_draftregistration AS D, auth_group G, auth_permission AS PERM
+    WHERE G.name = 'draft_registration_' || D.id || '_write'
+    AND (PERM.codename = 'read_draft_registration' OR PERM.codename = 'write_draft_registration');
+
+    -- Adds "read_draft_registration", "write_draft_registration", and "admin_draft_registration" permissions to all Draft Reg admin groups
+    INSERT INTO osf_draftregistrationgroupobjectpermission (content_object_id, group_id, permission_id)
+    SELECT D.id as content_object_id, G.id as group_id, PERM.id AS permission_id
+    FROM osf_draftregistration AS D, auth_group G, auth_permission AS PERM
+    WHERE G.name = 'draft_registration_' || D.id || '_admin'
+    AND (PERM.codename = 'read_draft_registration' OR PERM.codename = 'write_draft_registration' OR PERM.codename = 'admin_draft_registration');
+    """
+
+# Reverse migration - Remove all rows from DraftRegistrationGroupObjectPermission table - table gives draft django groups
+# permissions to draft reg
+drop_draft_reg_group_object_permission_table = """
+    DELETE FROM osf_draftregistrationgroupobjectpermission;
+    """
+
+def copy_editable_fields_from_draft_to_node(state, schema):
+    for draft_reg in DraftRegistration.objects.all():
+        node = draft_reg.branched_from
+        draft_reg.copy_editable_fields(node)
```

Comment:
This should probably be split off to a separate PR.  Copying the editable fields from the Node to the DraftRegistration for old drafts should happen in conjunction with new workflow release.

Reply:
split off into a separate ticket!